tinygrad/codegen/search.py

```diff
@@ -1,10 +1,23 @@
 from typing import Callable
 import time
+from functools import lru_cache
 from tinygrad.codegen.linearizer import Linearizer
 from tinygrad.helpers import DEBUG, prod, getenv
 
-UPCASTS = [1,2,3,4,5,6,7,8]
-LOCALS = [1,2,3,4,5,6,7,8,16,24,32]
+@lru_cache(maxsize=32)
+def get_divisors(n, min_div = 1, max_div = 512):
```

Comment:
If you want to generate list of divisors from given range, why not just doing this?

``` python
def get_divisors(n, min_div = 1, max_div = 512):
    for i in range(min_div, max_div + 1):
        if n % i == 0:
            yield i
```

Unless you want to use really really large numbers of max_div, you really don't have to overoptimize and overengineer with LRU cache.